src/smc-webapp/frame-editors/rmd-editor/actions.ts

```diff
@@ -2,30 +2,98 @@
 R Markdown Editor Actions
 */
 
+import { Set } from "immutable";
+import { debounce } from "lodash";
 import { Actions } from "../markdown-editor/actions";
-import { convert } from "./rmd2md";
+import { convert } from "./rmd-converter";
+import { markdown_to_html_frontmatter } from "../../markdown";
 import { FrameTree } from "../frame-tree/types";
+import { redux } from "../../app-framework";
+import { change_filename_extension } from "../generic/misc";
 
 export class RmdActions extends Actions {
+  private _last_save_time: number = 0;
+
   _init2(): void {
+    super._init2(); // that's the one in markdown-editor/actions.ts
     if (!this.is_public) {
       // one extra thing after markdown.
-      this._init_rmd2md();
+      this._init_rmd_converter();
+      this._check_produced_files();
     }
   }
 
-  _init_rmd2md(): void {
-    this._syncstring.on("save-to-disk", () => this._run_rmd2md());
-    this._run_rmd2md();
+  _init_rmd_converter(): void {
+    const run_debounced = debounce(() => this._run_rmd_converter(), 5 * 1000, {
```

Comment:
that's the lodash debounce, and well, it felt better that way because the whole mechanism is triggered by a save and it was behaving odd when I manually saved frequently. long term, I think we should have a build button and run-state indicator, similar to latex. Maybe worth a follow up ticket?